packages/mui-material/src/Autocomplete/Autocomplete.test.js

```diff
@@ -3001,6 +3001,26 @@ describe('<Autocomplete />', () => {
     });
   });
 
+  describe('prop: renderOption', () => {
+    it('should pass getOptionLabel through ownerState in renderOption callback', () => {
+      const { getByRole } = render(
+        <Autocomplete
+          open
+          options={[{ label: 'one' }]}
+          getOptionLabel={() => `getOptionLabel`}
+          renderInput={(params) => <TextField {...params} autoFocus />}
+          renderOption={function f(...[option, , , ownerState]) {
+            return <li key="key">{ownerState.getOptionLabel(option)}</li>;
+          }}
```

Comment:
You take a different property for the option:
```suggestion
          options={[{ name: 'Max' }]}
          getOptionLabel={(option) => option.name}
          renderInput={(params) => <TextField {...params} autoFocus />}
          renderOption={(props, option, optionState, ownerState) => <li data-id="option"> 
             {ownerState.getOptionLabel(option)}</li>; }
```